src/Microsoft.Extensions.ProjectModel.Sources/DotNet/DotNetProjectContext.cs

```diff
@@ -49,6 +50,10 @@ public DotNetProjectContext(ProjectContext projectContext, string configuration,
             // Workaround https://github.com/dotnet/cli/issues/3164
             IsClassLibrary = !(_compilerOptions.EmitEntryPoint
                     ?? projectContext.ProjectFile.GetCompilerOptions(null, configuration).EmitEntryPoint.GetValueOrDefault());
+
+            var dependencyProvider = new DotNetDependencyProvider(_projectContext);
```

Comment:
This forces project context to have a TFM (because of the null check). Can we move this out of ctor and only instantiate when required?